### PR TITLE
Alerting: Update documentation for MS Teams integration

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What is this feature?**
This updates documentation to configure MS Teams integration using the new recommended approach - Workflows. The current approach is deprecated https://devblogs.microsoft.com/microsoft365dev/retirement-of-office-365-connectors-within-microsoft-teams/

**Which issue(s) does this PR fix?**:

Related: https://github.com/grafana/grafana/issues/90139

- [ ] Decide on whether we want to add depreciation notice and refer to the MS blog post
- [ ] Decide on whether we want to add screenshots for how to create workflows

### CODE REVIEW

**Comment by yuri-tceretian** [comment]
@alyssawada do you think it makes sense to include some screenshots for how to create the workflow like in this comment https://github.com/grafana/grafana/issues/90139#issuecomment-2221510613?

**Comment by alyssawada** [comment]
> @alyssawada do you think it makes sense to include some screenshots for how to create the workflow like in this comment [#90139 (comment)](https://github.com/grafana/grafana/issues/90139#issuecomment-2221510613)?

@yuri-tceretian in general, our preference is to err on the side of not adding screenshots unless they add substantial value to the text. I'll defer to @brendamuir or someone closer to make that call for this case. But IMO, I'd go with no screenshots for now unless we receive some kind of feedback or request for clarity on this page.

File: docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-teams.md [hunk 1]
```diff
@@ -22,15 +22,13 @@ Use the Grafana Alerting - Microsoft Teams integration to receive notifications
...
+### Create a new Workflow in Microsoft Teams
 
-1. To add an incoming Webhook to a Teams channel, follow the steps in the [Create an Incoming Webhook guide](https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook?tabs=newteams%2Cdotnet#create-an-incoming-webhook).
-
-   Note: The steps to create incoming Webhooks may vary depending on the version of Microsoft Teams you are using. You can switch between the "New Teams" and "Classic Teams" tabs to view the relevant instructions for your version.
-
-1. Copy the Webhook URL.
+1. To create a new workflow follow the steps in [Create flows in Microsoft Teams](https://learn.microsoft.com/en-us/power-automate/teams/teams-app-create)
+2. Microsoft provides template library. You can use template "Post to a channel when a webhook request is received" 
+3. At the end of workflow creation wizard, copy the provided URL
```
Comment (alyssawada) [inline docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-teams.md:31]:
```suggestion
1. To create a new workflow, follow the steps in [Create flows in Microsoft Teams](https://learn.microsoft.com/en-us/power-automate/teams/teams-app-create) documentation. 
2. Consider using the template "Post to a channel when a webhook request is received" from the Microsoft template library. 
3. After creating the workflow in Teams, copy the provided URL for use in the next steps.
```

**Review by alyssawada — APPROVED** [review]
I suggested some minor editorial and stylistic changes but otherwise LGTM

File: docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-teams.md [hunk 2]
```diff
@@ -22,15 +22,13 @@ Use the Grafana Alerting - Microsoft Teams integration to receive notifications
 
 ## Before you begin
 
-To set up Microsoft Teams for integration with Grafana Alerting, you need to create an incoming Webhook. This allows Grafana to send alert notifications to Microsoft Teams channels.
+To set up Microsoft Teams for integration with Grafana Alerting, you need to create a new Workflow that accepts a webhook requests. This allows Grafana to send alert notifications to Microsoft Teams channels.
```
Comment (brendamuir) [inline docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-teams.md:25]:
```suggestion
To set up Microsoft Teams for integration with Grafana Alerting, create a new workflow that accepts Webhook requests. This allows Grafana to send alert notifications to Microsoft Teams channels.
```

File: docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-teams.md [hunk 3]
```diff
@@ -22,15 +22,13 @@ Use the Grafana Alerting - Microsoft Teams integration to receive notifications
...
 
-### Create an incoming Webhook in Microsoft Teams
+### Create a new Workflow in Microsoft Teams
```
Comment (brendamuir) [inline docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-teams.md:27]:
```suggestion
### Create a workflow in Microsoft Teams
```

File: docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-teams.md (hunk 1 above)
Comment (brendamuir) [inline docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-teams.md:31]:
```suggestion
1. To create a new workflow follow the steps in [Create flows in Microsoft Teams](https://learn.microsoft.com/en-us/power-automate/teams/teams-app-create)
1. Microsoft provides template library. You can use template "Post to a channel when a webhook request is received" 
1. At the end of workflow creation wizard, copy the provided URL
```

Comment (brendamuir) [inline docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-teams.md:31]:
```suggestion
1. To create a new workflow, follow the steps in [Create flows in Microsoft Teams](https://learn.microsoft.com/en-us/power-automate/teams/teams-app-create).
2. Microsoft provides a template library. You can use the template **Post to a channel when a webhook request is received**.
3. At the end of workflow creation wizard, copy the URL that is provided.
```

File: docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-teams.md [hunk 4]
```diff
@@ -22,15 +22,13 @@ Use the Grafana Alerting - Microsoft Teams integration to receive notifications
...
 
-1. To add an incoming Webhook to a Teams channel, follow the steps in the [Create an Incoming Webhook guide](https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook?tabs=newteams%2Cdotnet#create-an-incoming-webhook).
-
-   Note: The steps to create incoming Webhooks may vary depending on the version of Microsoft Teams you are using. You can switch between the "New Teams" and "Classic Teams" tabs to view the relevant instructions for your version.
-
-1. Copy the Webhook URL.
+1. To create a new workflow, follow the steps in [Create flows in Microsoft Teams](https://learn.microsoft.com/en-us/power-automate/teams/teams-app-create) documentation.
```
Comment (brendamuir) [inline docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-teams.md:29]: hey - we need to use repetitive list numbering - see here: https://grafana.com/docs/writers-toolkit/write/markdown-guide/#numbered-lists - so just 1. 1. 1. will render 1.2.3.

**Review by brendamuir — APPROVED** [review]

**Pull request merged**